### PR TITLE
Feat/antiflooding

### PR DESCRIPTION
**Pull request by sstanculeanu**

### CODE REVIEW

File: cmd/bridge/config/config.toml [hunk 1]
```diff
@@ -76,3 +76,13 @@
             SameSourceRequests = 10000
             # SameSourceResetIntervalInSec time frame between counter reset, in seconds
             SameSourceResetIntervalInSec = 1
+
+[TopicsAntiflood]
+    DefaultMaxMessagesPerInterval = 15000 # default number of messages per interval for a topic
```
Comment (iulianpascalau) [inline cmd/bridge/config/config.toml:81]: I think we can lower this to maybe 300

File: cmd/bridge/config/config.toml [hunk 2]
```diff
@@ -76,3 +76,13 @@
...
+	IntervalDurationInSeconds = 1 # interval duration for the allowed number of messages
+    [[TopicMaxMessagesConfig]]
+        Topic = "EthToElrond_join"
+        NumMessagesPerInterval = 500
```
Comment (iulianpascalau) [inline cmd/bridge/config/config.toml:85]: 100

File: cmd/bridge/config/config.toml [hunk 3]
```diff
@@ -76,3 +76,13 @@
...
+    [[TopicMaxMessagesConfig]]
+        Topic = "EthToElrond_sign"
+        NumMessagesPerInterval = 500
```
Comment (iulianpascalau) [inline cmd/bridge/config/config.toml:88]: 100

File: p2p/antiflood/topicFloodPreventer.go [hunk 4]
```diff
@@ -0,0 +1,79 @@
+package antiflood
```
Comment (iulianpascalau) [inline p2p/antiflood/topicFloodPreventer.go:1]: can we reuse the elrond-go/process/throttle/antiflood/topicFloodPreventer's code?

File: p2p/broadcaster.go [hunk 5]
```diff
@@ -113,6 +128,55 @@ func checkArgs(args ArgsBroadcaster) error {
 	return nil
 }
 
+func (b *broadcaster) createAntifloodHandler() (antiflood.AntifloodHandler, error) {
+	topicFloodPreventer, err := antiflood.NewTopicFloodPreventer(b.antifloodConfig.DefaultMaxMessagesPerInterval)
+	if err != nil {
+		return nil, err
+	}
+
+	antifloodHandler, err := antiflood.NewAntifloodHandler(topicFloodPreventer)
```
Comment (iulianpascalau) [inline p2p/broadcaster.go:137]: we also need to add some quota flood preventers. On elrond-go we are using 3 of them: fast reacting, slow reacting and absolute maximum one

File: p2p/broadcaster.go [hunk 6]
```diff
@@ -303,6 +374,9 @@ func (b *broadcaster) AddBroadcastClient(client core.BroadcastClient) error {
 
 // Close will close any containing members and clean any go routines associated
 func (b *broadcaster) Close() error {
+	if b.cancelResetAntifloodFunc != nil {
```
Comment (iulianpascalau) [inline p2p/broadcaster.go:377]: :+1:

File: testsCommon/p2p/antiflood/antifloodHandlerStub.go [hunk 7]
```diff
@@ -0,0 +1,38 @@
+package antiflood
+
+import "github.com/ElrondNetwork/elrond-go-core/core"
+
+type AntiFloodHandlerStub struct {
```
Comment (iulianpascalau) [inline testsCommon/p2p/antiflood/antifloodHandlerStub.go:5]: missing comment

**Comment by sstanculeanu** [comment]
closed branch and moved the changes here: https://github.com/ElrondNetwork/elrond-eth-bridge/pull/192

**Pull request closed without merging**